temporalio/worker/_workflow.py

```diff
@@ -268,6 +286,9 @@ async def _handle_activation(
                 self._could_not_evict_count += 1
                 return
 
+            if isinstance(err, DeadlockError) and err.has_pending_tb():
```

Comment:
```suggestion
            if isinstance(err, DeadlockError):
```

Just move this check into `swap_traceback` IMO (i.e. don't do anything if there is no pending traceback)